hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -339,6 +339,11 @@ def one_of(*args):  # noqa: F811
         # without incurring any performance overhead when there is only one
         # strategy, and keeps our reprs simple.
         return args[0]
+    if args and not any(isinstance(a, SearchStrategy) for a in args):
```

Comment:
Why is this `any` and not `all`?

Reply:
Because the idea is to only trigger this more-specific message if we think the user should be using `sampled_from()` instead of `one_of()` - the case where some args are valid and others are not is handled in the `OneOfStrategy` class.

Reply:
Ah, of course. Even though there weren't any double negatives in there I kind of got tripped up by a double negative anyway